polaris/curation/viz_utils.py

```diff
@@ -102,3 +140,200 @@ def check_undefined_stereocenters(dataset: PandasDataFrame):
     )
     fig = dm.to_image(to_plot.mol.tolist(), legends=legends)
     return fig
+
+
+def visualize_chemspace(
+    data: pd.DataFrame,
+    mol_col: str = "smiles",
+    size_col=None,
+    seed=1428,
+    umap_metric="jaccard",
+    hue_names: List[str] = None,
+):
+    """
+    Visualize the chemical space by doing a dimensionality reduction with UMAP.
+
+    Args:
+        data: a dataframe
+        split_names: names of the types of splits, should be found in columns of the "data" dataframe
+        mol_col: string column where we will look for a molecule to featurize
+        size_col: used for setting the style of the resulting scatterplot
+        seed: int, random seed to use for reproducibility
+        umap_metric: similarity metric to use for constructing the UMAP representation
+        hue_names: List of grouping variable that will produce points with different colors for the same UMAP plotting.
+
+    Returns:
+        figs, figures for each kind of split specified in split_names
+    """
+    figs = plt.figure(num=3)
+    features = [dm.to_fp(mol) for mol in data[mol_col]]  # Convert molecules to fingerprint
+    embedding = umap.UMAP(metric=umap_metric, random_state=seed).fit_transform(
+        features
+    )  # Embed the features with UMAP using a similarity metric
+    data["UMAP_0"], data["UMAP_1"] = embedding[:, 0], embedding[:, 1]
+
+    hue_names = [None] if hue_names is None else hue_names
+    for hue in hue_names:
+        plt.figure()
+        fig = sns.scatterplot(
+            data=data,
+            x="UMAP_0",
+            y="UMAP_1",
+            style=size_col,
+            hue=hue,
+            alpha=0.7,
+            palette="colorblind",
+        )
+        fig.set_title(f"UMAP embedding {hue}")
+    return figs
+
+
+def detailed_distributions_plots(
```

Comment:
This is a very nice function! Some bits of it seem reusable across multiple methods (e.g. for computing figure size). Not needed now, but seems we could extract some reusable code from this method to be used by other methods as well.